README.md

```diff
@@ -1,7 +1,108 @@
 # arugula - Scala Redis client wrapper for lettuce
 
-Arugula is a Redis client wrapper for Scala built on top of [Java lettuce](https://github.com/redis/lettuce)
+Arugula is a Redis client wrapper for Scala built on top of 
+[Java lettuce](https://github.com/redis/lettuce).
 
 ## Why another Scala Redis client
+There are already a number of Redis clients for Scala, including [scala-redis](https://github.com/debasishg/scala-redis) 
+and [scredis](https://github.com/scredis/scredis), so you might be wondering:
 
-The intent here is to introduce an alternative asynchronous Scala Redis client based on the [Java lettuce library](https://github.com/redis/lettuce) that simply leverages Scala Future
\ No newline at end of file
+Why create another one? 
+
+Maintaining a redis client is a lot of work, and the existing clients are either poorly maintained
+or have heavy dependencies like Akka or Cats-Effect. We were looking for a clean Scala API with 
+minimal dependencies that would allow us to use the full feature set of Redis and that could be easily tested.
+
+[Lettuce](https://github.com/redis/lettuce) is a well-maintained and feature-rich Redis client for Java. 
+While it is possible to use the Java client directly in Scala,
+Arugula provides a more idiomatic Scala API and is designed to be lightweight and easy to use.
+Rather than trying to reinvent the wheel, Arugula leverages the work done by the [Lettuce](https://github.com/redis/lettuce) team 
+and provides a more Scala-friendly API. 
+All the commands map directly to the Lettuce commands, so the API is very similar to the Lettuce API.
+
+## Features
+The code is split up into two main packages:
+- `arugula-api` - contains the pure scala API
+- `arugula-core` - contains the bindings to Lettuce
+
+The api allows you to create a Redis client and execute commands on it. 
+The core package contains the implementation of the commands using Lettuce.
+An implementation can be written independently of Lettuce, and the commands can be tested 
+without a Redis Server or the Lettuce dependency.
+
+## Usage
+### SBT Configuration
+To use Arugula, add the following to your `build.sbt` file:
+
+
+```scala
+val arugulaVersion = "0.0.1"
+
+libraryDependencies += "com.github.scoquelin" %% "arugula-api" % arugulaVersion
+libraryDependencies += "com.github.scoquelin" %% "arugula-core" % arugulaVersion
+```
+
+This will add the Arugula API and core packages to your project along with the Lettuce dependency.
+If you want to only use the API in your package, you can exclude the core package from the dependencies
+and code only against the API. This decouples the API from the core implementation and allows you to write
+your own implementation of the commands or write an adapter for another underlying Redis client.
+
+In the future we could add support for a Jedis wrapper or other JVM-based Redis clients. That said,
```

Comment:
Not sure about that part, Jedis does not support asynchronous call and they dropped the ball for supporting it : https://github.com/redis/jedis/pull/713

Reply:
k. will remove that comment.

Reply:
I could just say:

> In the future we could add support for other JVM-based Redis clients.

Reply:
got rid of that part and re-worded it to just focus on using Lettuce documentation to provide more insight on how to use Arugula.